### DISCUSSION TITLE
Encryption defaults

### DISCUSSION THREAD

**Discussion by riccardone** [icsharpcode/SharpZipLib#518] [General] [answered]
Hi,

When I set the password on the FastZip class, could you please confirm what is the default Algorithm (AES or ?) and if AES is used what is the default AESKeySize value?

Many thanks

**Comment by piksel** [comment, marked as answer]
It defaults to [ZipCrypto](https://github.com/icsharpcode/SharpZipLib/blob/master/src/ICSharpCode.SharpZipLib/Zip/FastZip.cs#L236), which is the original (weak) encryption. AES encryption for FastZip was added in #380, which will be included in v1.3 (not yet released).

**Reply by riccardone** [reply]
Ok thanks for confirming. 

While waiting the next version I copied this class that works pretty well
https://github.com/icsharpcode/SharpZipLib/discussions/454#discussioncomment-46390 
 
In the next version I would suggest defaults to AES 256 in case you just set the password. 
Also mark ZipCrypto as depracated could be something to consider.

**Reply by piksel** [reply]
While I do agree that ZipCrypto should be avoided except for compability reasons (but even that is hardly relevant anymore), we don't want to make breaking changes to the API, so changing the default is not possible.
